src/Persistence/SlotEntityContentRepository.php

```diff
@@ -55,10 +58,18 @@ public function setContent( EntityId $entityId, Content $content ): void {
 		$page = $this->getWikiPage( $entityId );
 
 		if ( $page !== null ) {
+			$this->checkPageUserPermissions( $page );
 			$updater = $page->newPageUpdater( $this->authority );
 			$updater->setContent( $this->slotName, $content );
 			$updater->saveRevision( CommentStoreComment::newUnsavedComment( 'Updated entity mappings' ) ); // TODO
 		}
 	}
 
+	private function checkPageUserPermissions( WikiPage $page ): void {
+		$status = new PermissionStatus();
+		if ( !$this->authority->authorizeWrite( 'edit', $page, $status ) ) {
```

Comment:
I didn't add the 'create' permission because you cannot create mappings before the entity page exists.